src/components/data-points/data-points.tsx

```diff
@@ -107,72 +69,55 @@ export default function DataPoints(props: DataPointProps) {
       },
     })
 
-  const buildState = useCallback(
-    (dataPoints: DataPointType[][]): DataPointsState => {
-      const combinedVariables: CombinedVariableType[] = buildCombinedVariables()
-      const dataPointRows: TableDataRow[] = dataPoints
-        .map(item => {
-          const rowData: DataPointType[] = item.filter(
-            dp => !scoreNames.includes(dp.name)
-          )
-          const vars: TableDataPoint[] = new Array(rowData.length)
-          rowData.forEach(v => {
-            const idx = combinedVariables.findIndex(it => it.name === v.name)
-            vars[idx] = {
-              name: v.name,
-              value: v.value.toString(),
-              options: combinedVariables[idx]?.options,
-              tooltip: combinedVariables[idx]?.tooltip,
-            }
-          })
-          const scores: TableDataPoint[] = item
-            .filter(dp => scoreNames.includes(dp.name))
-            .map(score => ({ name: score.name, value: score.value as string }))
-          return {
-            dataPoints: vars.concat(scores),
-            isNew: false,
-          }
-        })
-        .concat(buildEmptyRow())
-
-      return {
-        rows: dataPointRows,
-        changed: false,
-      }
-    },
-    [buildCombinedVariables, scoreNames, buildEmptyRow]
-  )
-
   useEffect(() => {
-    dispatch({ type: 'setInitialState', payload: buildState(dataPoints) })
-  }, [valueVariables, categoricalVariables, scoreNames, buildState, dataPoints])
+    dispatch({
+      type: 'setInitialState',
+      payload: {
+        valueVariables,
+        categoricalVariables,
+        scoreNames,
+        data: dataPoints,
+      },
+    })
+  }, [valueVariables, categoricalVariables, scoreNames, dataPoints])
 
   useEffect(() => {
-    const updateDataPoints = (dataRows: TableDataRow[]) => {
+    const convertEditableRowToExperimentRow = (
+      row: TableDataRow | undefined
+    ) => {
+      if (row === undefined) {
+        return []
+      }
+      const vars = row.dataPoints.filter(dp => !scoreNames.includes(dp.name))
+      const scores = row.dataPoints
+        .filter(dp => scoreNames.includes(dp.name))
+        .map(s => ({
+          name: s.name,
+          value: parseFloat(s.value),
+        }))
+      return vars
+        .map(dp => ({
+          name: dp.name,
+          value: dp.value as DataPointTypeValue,
+        }))
+        .concat(scores) as DataEntry['data']
+    }
+    const updateDataPoints = (
+      meta: DataEntry['meta'][],
```

Comment:
Is ```DataEntry['meta'][]``` this the same as ```DataEntryMetaData[]``` and could/should that be used instead? Would require it to be exported from common.ts